dd-sdk-android-gradle-plugin/src/main/kotlin/com/datadog/gradle/plugin/DdFileUploadTask.kt

```diff
@@ -0,0 +1,326 @@
+package com.datadog.gradle.plugin
+
+import com.android.build.gradle.api.ApplicationVariant
+import com.datadog.gradle.plugin.internal.ApiKey
+import com.datadog.gradle.plugin.internal.ApiKeySource
+import com.datadog.gradle.plugin.internal.DdAppIdentifier
+import com.datadog.gradle.plugin.internal.OkHttpUploader
+import com.datadog.gradle.plugin.internal.Uploader
+import org.gradle.api.DefaultTask
+import org.gradle.api.logging.Logging
+import org.gradle.api.provider.Provider
+import org.gradle.api.provider.ProviderFactory
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.TaskAction
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+import java.io.File
+import javax.inject.Inject
+
+/**
+ * A Gradle task to upload symbolication files to Datadog servers (NDK symbol files,
+ * Proguard/R8 files, etc.)..
+ */
+abstract class DdFileUploadTask @Inject constructor(
+    providerFactory: ProviderFactory
+) : DefaultTask() {
+    @get:Internal
+    internal var uploader: Uploader = OkHttpUploader()
+
+    /**
+     * The API key to use for uploading.
+     */
+    @get:Input
+    var apiKey: String = ""
+
+    private val disableGzipOption: Provider<String> =
+        providerFactory.gradleProperty(DdFileUploadTask.DISABLE_GZIP_GRADLE_PROPERTY)
+
+    /**
+     * Source of the API key set: environment, gradle property, etc.
+     */
+    @get:Input
+    var apiKeySource: ApiKeySource = ApiKeySource.NONE
+
+    /**
+     * The repository detector to use for detecting the repository information.
+     */
+    @get:Internal
+    var repositoryDetector: RepositoryDetector? = null
```

Comment:
Would need to go through the service provider since we're using `register` and not constructing the object directly. I meant to look back into it and forgot. I'll see what I can see.

Reply:
You can still provide this value in the following [register](https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/TaskContainer.html#register-java.lang.String-java.lang.Class-java.lang.Object...-) method overload the same way it is done for `create`, so value is injected in the constructor and you don't have a nullable type here.